Dynamic import glitches: Cannot find module

Running the latest beta, `1.8.1-beta.20` from #10248 , I'm working on upgrading a fairly large app to benefit from dynamic imports. Part of the process is dynamically importing templates at route-level, then importing all used child templates in the dynamically imported route-level templates. Like so:

```handlebars
<!-- parent.html -->
<template name="parent">
  {{> child ''}}
</template>
```

```js
// parent.js

import './child';

Template.parent....
```

```js
// routes
FlowRouter.route('/', {
  name: 'landing',
  waitOn() {
    return import('../../client/parent');
  },
  action() {
    this.render('parent');
  }
});
```

### Problem

For some reason, when adding new imports, I'm first getting `Cannot find module`error thrown, even though the path is perfectly valid.

```
[ostrio:flow-router-extra] [route.wait] Promise not resolved Error: Cannot find module '../../../components/tool_icons/mailbox_icon/mailbox_icon'
    at makeMissingError (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:232)
    at Module.require (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:251)
    at Module.moduleLink [as link] (modules.js?hash=09b63e0002909144323322723ef16c87833f772d:371)
    at eval (app_menu.js:1)
    at dynamic-import.js?hash=76097fc6a86e6a6d4b974469a5ee8a6a8ec2e694:138
    at fileEvaluate (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:346)
    at Module.require (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:248)
    at Module.moduleLink [as link] (modules.js?hash=09b63e0002909144323322723ef16c87833f772d:371)
    at eval (landing.js:1)
    at dynamic-import.js?hash=76097fc6a86e6a6d4b974469a5ee8a6a8ec2e694:138
```

### Workaround

1. Restart the Meteor app, might help...
2. ...If not, move the import declaration to a different line on the same file
-> Module is found and the error is gone.

This however, is painfully slow, having to restart the entire app all the time.

Once again I was too hasty in reporting the issue, since right after I discovered more information. I just noticed, that I can make the module to be **not found** simply by removing the empty line between two imports.

From...

```
import './components/app_nav/app_nav';

import './components/header_right_menu/header_right_menu';
```

into this...

```
import './components/app_nav/app_nav';
import './components/header_right_menu/header_right_menu';
```

...results in `modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:232 Uncaught (in promise) Error: Cannot find module './components/app_nav/app_nav'`

Saga continues: it also happens with Meteor packages. Only deleting `.meteor/local` helped 😫

```
[ostrio:flow-router-extra] [route.wait] Promise not resolved Error: Cannot find module '/node_modules/meteor/arggh:dates/dates.js'
    at makeMissingError (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:232)
    at fileEvaluate (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:335)
    at Module.require (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:248)
    at Module.moduleLink [as link] (modules.js?hash=0190bb44e52acc3c8893785a5abdc460c640d217:371)
    at eval (date_banner.js:1)
    at dynamic-import.js?hash=4c6f5c174cf2e22c6c3f680ed4c5ed9c7f9a921c:138
    at fileEvaluate (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:346)
    at Module.require (modules-runtime.js?hash=d3c3e5d67c95f97a60888bda7373292efad3be5e:248)
    at Module.moduleLink [as link] (modules.js?hash=0190bb44e52acc3c8893785a5abdc460c640d217:371)
    at eval (landing.js:1)
```

--- a/tools/isobuild/linker.js
+++ b/tools/isobuild/linker.js
@@ -26,11 +26,6 @@ const DYNAMIC_PRELINKED_OUTPUT_CACHE = new LRU({
   max: Math.pow(2, 11)
 });
 
-// Caches main bundle for client architectures that have dynamic imports
-const PRELINKED_MAIN_CACHE = new LRU({
-  max: Math.pow(2, 3)
-});
-
 var packageDot = function (name) {
   if (/^[a-zA-Z][a-zA-Z0-9]*$/.exec(name)) {
     return "Package." + name;
@@ -214,54 +209,6 @@ _.extend(Module.prototype, {
       });
     }
 
-    // Caching is only useful when there are dynamic files.
-    const canCache = self.useGlobalNamespace && results.find(result => result.dynamic);
-    const key = `${self.bundleArch}-${self.name}`;
-    let inputHash;
-
-    // When only dynamic files changed, reuse the main result from
-    // the last time it was generated
-    if (canCache) {
-      const files = [];
-
-      self.files.forEach(file => {
-        if (file.bare || !file.isDynamic()) {
-          files.push({
-            hash: file._inputHash,
-            meteorInstallOptions: file.meteorInstallOptions,
-            sourceMap: !!file.sourceMap,
-            mainModule: file.imported,
-            alias: file.alias,
-            lazy: file.lazy,
-            bare: file.bare
-          });
-        }
-      });
-
-      inputHash = watch.sha1(JSON.stringify({
-        useGlobalNamespace: this.useGlobalNamespace,
-        combinedServePath: this.combinedServePath,
-        name: self.name,
-        files
-      }));
-
-      // By using a more general key only the latest result is stored
-      // The inputHash is used to know when the cache is stale
-      if (PRELINKED_MAIN_CACHE.has(key)) {
-        let {
-          inputHash: cachedInputHash,
-          result: cachedResult
-        } = PRELINKED_MAIN_CACHE.get(key);
-
-        if (cachedInputHash === inputHash) {
-          result.source = cachedResult.source;
-          result.sourceMap = cachedResult.sourceMap;
-
-          return results;
-        }
-      }
-    }
-
     var node = new sourcemap.SourceNode(null, null, null, chunks);
 
     Profile.time(
@@ -284,13 +231,6 @@ _.extend(Module.prototype, {
         }
       }
     );
-
-    if (canCache) {
-      PRELINKED_MAIN_CACHE.set(key, {
-        inputHash,
-        result
-      });
-    }
 
     return results;
   }),

